### PR TITLE
Fix generic defaulting w/ middleware types

### PR DESCRIPTION
**Pull request by petejohanson**
* Need to put the `TEntities` first in the generic parameters,
  so the action generic type can be defaulted properly.

### CODE REVIEW

File: packages/redux-query/index.d.ts [hunk 1]
```diff
@@ -254,7 +254,7 @@ declare module 'redux-query' {
   };
 
   export const queryMiddleware: QueryMiddlewareFactory;
-  export interface ReduxQueryDispatch<A extends AnyAction = ReduxQueryAction<TEntities>, TEntities = Entities> {
+  export interface ReduxQueryDispatch<TEntities = Entities, A extends AnyAction = ReduxQueryAction<TEntities>> {
```
Comment (rctbusk) [inline packages/redux-query/index.d.ts]: Why the change in order here?

Comment (macjohnny) [reply packages/redux-query/index.d.ts]: Otherwise you get this errror:

node_modules/redux-query/index.d.ts:257:78 - error TS2744: Type parameter defaults can only reference previously declared type parameters.

257   export interface ReduxQueryDispatch<A extends AnyAction = ReduxQueryAction<TEntities>, TEntities = Entities> {


See also https://github.com/OpenAPITools/openapi-generator/pull/3824

**Comment by petejohanson** [comment]
@rctbusk Thanks for merging this! Any chance we can get a 3.2.1 release that publishes this?

**Comment by rctbusk** [comment]
Yep! Working on testing the patch now

**Comment by rctbusk** [comment]
I have published an alpha release and am testing it now. https://www.npmjs.com/package/redux-query
